crates/bitwarden-core/src/auth/key_connector.rs

```diff
@@ -0,0 +1,42 @@
+use bitwarden_crypto::{CryptoError, MasterKey, RsaKeyPair};
+
+#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
+pub struct KeyConnectorResponse {
+    pub master_key: String,
+    pub encrypted_user_key: String,
+    pub keys: RsaKeyPair,
+}
+
+pub(super) fn make_key_connector_keys(
+    mut rng: impl rand::RngCore,
+) -> Result<KeyConnectorResponse, CryptoError> {
+    let master_key = MasterKey::generate(&mut rng);
+    let (user_key, encrypted_user_key) = master_key.make_user_key()?;
+    let keys = user_key.make_key_pair()?;
+
+    Ok(KeyConnectorResponse {
+        master_key: master_key.to_base64(),
+        encrypted_user_key: encrypted_user_key.to_string(),
+        keys,
+    })
```

Comment:
This is quite different compared to our current approach, but helps avoiding the weird steps in our ts project @jlf0dev.

Reply:
Perhaps I don't understand the key connector flow, but I had thought the encrypting keys were generated by key connect -- are they just stored there?

Reply:
Correct - just stored there.